### DISCUSSION TITLE
Maps:  text alternative that serves the equivalent purpose

### DISCUSSION THREAD

**Discussion by bruce-usab** [atbcb/ICTTestingBaseline#487] [Ideas]
What could/should an auditor require for when evaluating text alternatives for maps?  When is a description barely sufficient?  What would it take for the text alternative to serve the equivalent purpose?

For the purposes of this discussion, please assume a static image and a static web page.

### AccessiblePublishing.ca

[Guide to Writing Long Descriptions](https://www.accessiblepublishing.ca/guide-to-long-descriptions/)

> **Maps**
> - Include the Name/Title of the map and include a description of the legend.
> - When describing the legend, consider its purpose, and only describe important information.
> - Aim for clarity, even if detail is sacrificed. Focus on the information that is relevant in the greater context of the book.
> - Ask yourself: What is this map telling the reader? To help answer this question, review the context surrounding the map.

### NBA Tape Recording Manual

[Excerpts from the NBA Tape Recording Manual, Third Edition](https://www.w3.org/2000/08/nba-manual/)
How to describe a figure:
> 1. Ask yourself, "Why is it there? What does it illustrate or add to the text?" Stress these points and avoid inconsequential details.
> 1. Consider the grade level of the text in choosing the words you will use in the description. Make comparisons to items with which a blind person will be familiar, e.g., the hand, a baseball, a triangle and the like.
> 1. First describe the illustration as a whole. Don't talk about a window until you have said the illustration is a picture of a house.
> 1. Pick out a few reference points and locate them within the figure. Describe the details in a logical sequence, positioning them in relation to the reference points.

Second example in the section is a grayscale map (and description) of [European Claims in North America in 1650](https://www.w3.org/2000/08/nba-manual/map.gif).

### Diagram Center

[Specific Guidelines – Maps](http://diagramcenter.org/specific-guidelines-e-2.html)
> 1. Geographic Maps
> - Maps may be decorative, or simple locator maps, and may only require a short description. See Description 1.
> - If the map is used to illustrate a concept and help define new terms, describe the lines of latitude and longitude and explain how they look on the map (horizontal, vertical), and the corresponding degrees. See Description 2.
> - If the map is used as part of a geography lesson and the land and water are relevant, describe the labeled areas, including their relationship and distance to each other when relevant. See Description 3.
>
> 2. Political Maps
> - Focus on the map’s central teaching point to determine if borders, region shapes, and bodies of water are important.
> - If the map is an essential part of the lesson or assessment, provide a general overview along with details and place names. Include labeled landmarks and labeled bodies of water in the description. If there are too many labels (if all 50 states are labeled on a map of the U.S., for example), focus on the labels that are most relevant to the figure and the concept it is depicting. Whenever possible, organize the description using bulleted lists and by pulling the most important information to the beginning of the description so students hear it first. See Description 1.
> - If the map is supporting surrounding text, or if it has a detailed caption, describe general trends in the map and refer to large areas at once. See Description 2.
> - The elements in the key can be folded in to the description of the map itself and do not need to be described separately.

Two examples (one geographic, one political) are included, along with two or three descriptions for each.

### More Resources

These are new to me (Bruce) and authoritative.  I could not find guidance for maps specifically.

- [Cooper Hewitt Guidelines for Image Description](https://www.cooperhewitt.org/cooper-hewitt-guidelines-for-image-description/)
- WGBH NCAM [Guidelines for Describing STEM Images](https://www.wgbh.org/guidelines-for-describing-stem-images)
- Perkins School for the Blind [How to Write Alt Text and Image Descriptions for the visually impaired](https://www.perkins.org/resource/how-write-alt-text-and-image-descriptions-visually-impaired) "Fully updated July 2023"

**Comment by bruce-usab** [comment]
From recent conversation, and the perspective of an auditor (as opposed to author), I am wondering if the best we can do is a list of obvious omissions?   WRT a text alternative for a map, it almost always possible to find a question which is answered by the map, but not answered by the text alternative.

**Reply by bruce-usab** [reply]
Cliff noted that their alt will never be enough – a tester can always find a shortcoming.

**Comment by kengdoj** [comment]
To be specific, this is about a map that can be used for navigation (and not graphical representations of data where an equivalent alternative could be a data table).

**Comment by kengdoj** [comment]
To stir discussion, here's a college campus example on a "Visit Us" page. Imagine all the buildings have labels, library, dorms, student center, admin buildings, parking, etc.

![TBD](https://github.com/atbcb/ICTTestingBaseline/assets/21043936/2f960256-103a-450e-a438-4dead25df6c0)



Would this pass? `alt= "map of campus with visitor check-in marked with a big X"`. If not, how would you improve it? And how could the Baseline require such improvement?

**Comment by michaelhortongsa** [comment]
Solving only for non-visual users, my overarching thought is that description is equivalent to intent; which in this example is meet: me at this location. Consolidating my thoughts, I think the image fails to provide a point "a" which needed to give directions to point "x". Adding point "a" would allow the author, who intends to indicate the location of registration area within a campus so that people can find that location, a common reference to then be able to park and walk to point "b". The tester, not knowing the campus, would then have sufficient information to locate point "b" in real life. Perhaps the test would require point "a" - in a "directions to a location" map - also be a place where a human is located to get assistance?

**Comment by KMSOC** [comment]
Alt text should follow the guiding principle for any image: If I removed the map and just had the information in the document and the alt text, would all the critical information being communicated by that map still be fully available? What critical information would I lose if I couldn’t perceive the map? If there is information loss without the image, there may be more work to do to ensure that the map is accessible to everyone. In the way we have our image test now, we defer to the author as to the intent. 

Our challenge is still related to complex maps that are meant to be exploratory or navigational. In these cases, I would suggest attempting to understand the author's intention. If the author did not intend the map to be used in any way possible, it should be acceptable to have alt text that matches the author's intention (also understand that the tester may not be able to know the author's intention so this is still complicated). Thoughts:

-   Required: An additional text alternative for **all** relevant data and information (again deferring to author's intention)-limitation is that this will not always be possible. 
        - This is more directly applicable in the case of maps that are visual representation of data tables (e.g., https://www.cdc.gov/wcms/4.0/cdc-wp/data-presentation/examples/example-us-map-counties.html)
        - This **may not** be sufficient for something such as a campus map/navigational map/exploratory map. I believe it may be acceptable to fail complex maps in which the only way to provide full equivalent information is via an accessible interactive alternative.
        - That being said, going back to the very first paragraph, if I were to remove the image of the map above and simply provide 4 different scenarios with how to get to the check in, I would be hard-pressed to fail that. 
        - Not the best example for accessibility as a whole (ignore the glaring issues and focus on the map and text) but see these directions to get to parking for Charlotte RenFest. I would argue that the text instructions are pretty sufficient for a variety of use cases. This is also a scenario in which Google Maps is unreliable, so this is THE map to get to RenFest: https://static1.squarespace.com/static/5fd0f430d10d883954d0cb63/t/62b9c4a521abac4daae23824/1656341669266/RenFest+Recommended+Driving+Directions.pdf

In sum- I support deferring to authoring intent but also support a fail of a map where it is determined that text (alt or otherwise) alone (no matter what) is insufficient to provide fully equivalent info.

**Comment by kengdoj** [comment]
> Would this pass? `alt= "map of campus with visitor check-in marked with a big X"`. If not, how would you improve it? 

I would not pass it unless there was page text that included useful landmarks near the check-in location so that the map is just supplemental. Something like: The visitor check-in is by the fountain in front of [building name].

If that info was not in page text, and the map is the sole source of the check-in location, then I would require more info in the alternative text of the map: `alt= "map of campus with visitor check-in location by the fountain in front of [building name] marked with a big X"`

> And how could the Baseline require such improvement?

Maybe under the [6. Images, Advisory](https://ictbaseline.access-board.gov/06Images/#advisory-tips-for-streamlined-test-processes): An image that conveys substantial details, relationships, locations, etc., must include such information in the text alternative to provide the equivalent purpose of the image. See [WAI Complex Images](https://www.w3.org/WAI/tutorials/images/complex/) for examples.

**Comment by kengdoj** [comment]
Determine the purpose of the map: 

|Map's purpose(s) | Map's text alternative must - |
|--------| ---------|
| Navigation | Identify the destination indicated on the map and provide step-by-step directions from one or more landmarks, including starting point, turns, and notable waypoints. |
| Display location of objects | Describe the spatial relationships between relevant labeled objects or landmarks, including their relative positions, proximity, and orientation to one another |

Additional Requirements
---------
The text alternative of a map must include information from the map’s legend, if one is present, to explain the meaning of symbols, colors, or patterns used.

**Comment by kengdoj** [comment]
Additional requirement: information conveyed visually, such as trends, groupings, comparisons, density, etc. must be included in alternative text to serve equivalent purpose.
